src/main/java/com/github/tomakehurst/wiremock/http/RequestMethodMatcher.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2023-2024 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tomakehurst.wiremock.http;
+
+import com.github.tomakehurst.wiremock.matching.MatchResult;
+import com.github.tomakehurst.wiremock.matching.NamedValueMatcher;
+import java.util.Set;
+
+public class RequestMethodMatcher implements NamedValueMatcher<Set<RequestMethod>> {
+
+  private final Set<RequestMethod> requestMethods;
+
+  public RequestMethodMatcher(Set<RequestMethod> requestMethods) {
```

Comment:
It would be great if when serialised this was wrapped in a `oneOf` element in a similar fashion to other matchers, a) because this communicates the intent more clearly to the reader and b) permits us to e.g. add `noneOf` in the future.

Reply:
Unfortunately, I don't quite understand(
Maybe we’ll leave it like this for now?))) and change it when needed?)))

Reply:
What I mean is instead of
```
"method": ["GET", "POST"]
```

We'd do
```
"method": {
  "oneOf": ["GET", "POST"]
}
```

That way we give ourselves the option of putting another operator there if we ever need to. The problem with waiting until later to do that is that it'd then be a breaking change.

Reply:
I managed to do it like oneOfRequestMethods: [methods]
I may add noneOfRequestMethods: [methods]

Reply:
I do "isNoneOf" right away )

Reply:
@tomakehurst
I hope this is the final version)
Please look)